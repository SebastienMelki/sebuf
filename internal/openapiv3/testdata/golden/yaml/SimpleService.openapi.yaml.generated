openapi: 3.1.0
info:
    title: SimpleService API
    version: 1.0.0
paths:
    /SimpleService/Create:
        post:
            tags:
                - SimpleService
            summary: Create
            description: Create a simple resource
            operationId: Create
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/SimpleRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/SimpleResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /SimpleService/Get:
        post:
            tags:
                - SimpleService
            summary: Get
            description: Get a simple resource
            operationId: Get
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/SimpleRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/SimpleResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        FieldViolation:
            type: object
            properties:
                field:
                    type: string
                    description: The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')
                description:
                    type: string
                    description: Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')
            required:
                - field
                - description
            description: FieldViolation describes a single validation error for a specific field.
        ValidationError:
            type: object
            properties:
                violations:
                    type: array
                    items:
                        $ref: '#/components/schemas/FieldViolation'
                    description: List of validation violations
            required:
                - violations
            description: ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.
        SimpleRequest:
            type: object
            properties:
                name:
                    type: string
                    description: User's name
                age:
                    type: integer
                    format: int32
                    description: User's age
                active:
                    type: boolean
                    description: Whether user is active
            description: Simple message for testing basic OpenAPI generation
        SimpleResponse:
            type: object
            properties:
                message:
                    type: string
                    description: Response message
                success:
                    type: boolean
                    description: Success indicator
                timestamp:
                    type: integer
                    format: int64
                    description: Response timestamp
            description: Simple response message
