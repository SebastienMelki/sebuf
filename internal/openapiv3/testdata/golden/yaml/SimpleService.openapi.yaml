openapi: 3.1.0
info:
    title: SimpleService API
    version: 1.0.0
paths:
    /SimpleService/Create:
        post:
            tags:
                - SimpleService
            summary: Create
            description: Create a simple resource
            operationId: Create
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/SimpleRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/SimpleResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /SimpleService/Get:
        post:
            tags:
                - SimpleService
            summary: Get
            description: Get a simple resource
            operationId: Get
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/SimpleRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/SimpleResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        SimpleRequest:
            type: object
            properties:
                name:
                    type: string
                    description: User's name
                age:
                    type: integer
                    format: int32
                    description: User's age
                active:
                    type: boolean
                    description: Whether user is active
            description: Simple message for testing basic OpenAPI generation
        SimpleResponse:
            type: object
            properties:
                message:
                    type: string
                    description: Response message
                success:
                    type: boolean
                    description: Success indicator
                timestamp:
                    type: integer
                    format: int64
                    description: Response timestamp
            description: Simple response message
